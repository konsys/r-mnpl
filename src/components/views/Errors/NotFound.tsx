import { Button, Grid, Hidden, Typography } from "@material-ui/core";

import { GRID_SPACING } from "../../../theme";
import { Helmet } from "react-helmet";
import { ReactComponent as NotFoundIcon } from "../../../theme/svg/404.svg";
import React from "react";
import Template from "../Template/Template";
import { useTranslation } from "react-i18next";

export default function NotFound() {
  const { t } = useTranslation();
  const component = (
    <Grid
      container
      alignItems="center"
      justify="center"
      direction="column"
      spacing={GRID_SPACING}
    >
      <Grid item style={{ height: "25%", width: "50%" }}>
        <NotFoundIcon />
      </Grid>
      <Hidden smDown>
        <Grid item>
          <Typography variant="h4" gutterBottom>
            {t("Page not found")}
          </Typography>
        </Grid>
      </Hidden>
      <Hidden mdUp>
        <Grid item>
          <Typography variant="body1" gutterBottom>
            {t("Page not found")}
          </Typography>
        </Grid>
      </Hidden>

      <Hidden smDown>
        <Grid item>
          <Typography variant="body1">
            {t("This page doesn`t exist yet")}
          </Typography>
        </Grid>{" "}
      </Hidden>
      <Hidden mdUp>
        <Grid item>
          <Typography variant="body2">
            {t("This page doesn`t exist yet")}
          </Typography>
        </Grid>{" "}
      </Hidden>
      <Grid item>
        <Button size={"medium"} variant="contained" color="primary">
          <Typography variant="body2" noWrap>
            {t("To main page")}
          </Typography>
        </Button>
      </Grid>
    </Grid>
  );
  return (
    <>
      <Helmet>
        <title>{t("Page not found")}</title>
      </Helmet>
      <Template columns={1}>{component}</Template>
    </>
  );
}
