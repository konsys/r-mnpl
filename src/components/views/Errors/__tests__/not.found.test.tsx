import NotFound from "../NotFound";
import React from "react";
import { shallow } from "enzyme";

describe("Tokens test", () => {
  it("should render", () => {
    expect(shallow(<NotFound />)).toMatchSnapshot();
  });
});
