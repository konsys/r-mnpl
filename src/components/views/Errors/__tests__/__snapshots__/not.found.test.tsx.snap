// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens test should render 1`] = `ShallowWrapper {}`;
